pandas/tests/extension/base/ops.py

```diff
@@ -0,0 +1,66 @@
+import pytest
+import numpy as np
+import pandas as pd
+from .base import BaseExtensionTests
+
+
+class BaseOpsTests(BaseExtensionTests):
+    """Various Series and DataFrame ops methos."""
+
+    def check_op(self, s, op, other, exc=NotImplementedError):
+
+        with pytest.raises(exc):
+            getattr(s, op)(other)
```

Comment:
IMO we should rather test `op(s, other)`, but this is because the fixture gives the dunder ops?

Reply:
Ah, I see you do that below for the decimal tests. 
But, I might maybe rather put that implementation here as the default, and overwrite it in categorical to check it raises instead?

Reply:
Hmm, although, then the json tests also needs to skip it. Maybe we can have both versions here in the base tests, and then you can subclass the appropriate one (I don't know how typical it would be external authors would be able to reuse the base class for those tests though)

Reply:
@jorisvandenbossche I copied this from implementation done by @jreback elsewhere.  I actually agree with you that we should use the op here rather than the dunder name.  I will change the base class and see how things look.